fhir-audit/src/main/java/com/ibm/fhir/audit/logging/mapper/impl/AuditEventMapper.java

```diff
@@ -0,0 +1,474 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.audit.logging.mapper.impl;
+
+import static com.ibm.fhir.model.type.Code.code;
+import static com.ibm.fhir.model.type.String.string;
+import static com.ibm.fhir.model.type.Uri.uri;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+import com.ibm.fhir.audit.logging.api.AuditLogEventType;
+import com.ibm.fhir.audit.logging.beans.AuditLogEntry;
+import com.ibm.fhir.audit.logging.beans.FHIRContext;
+import com.ibm.fhir.audit.logging.mapper.Mapper;
+import com.ibm.fhir.config.PropertyGroup;
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.generator.FHIRGenerator;
+import com.ibm.fhir.model.resource.AuditEvent;
+import com.ibm.fhir.model.resource.AuditEvent.Agent;
+import com.ibm.fhir.model.resource.AuditEvent.Agent.Network;
+import com.ibm.fhir.model.resource.AuditEvent.Entity;
+import com.ibm.fhir.model.resource.AuditEvent.Entity.Detail;
+import com.ibm.fhir.model.resource.AuditEvent.Source;
+import com.ibm.fhir.model.type.Base64Binary;
+import com.ibm.fhir.model.type.CodeableConcept;
+import com.ibm.fhir.model.type.Coding;
+import com.ibm.fhir.model.type.Instant;
+import com.ibm.fhir.model.type.Reference;
+import com.ibm.fhir.model.type.code.AuditEventAction;
+import com.ibm.fhir.model.type.code.AuditEventAgentNetworkType;
+
+/**
+ * This class adds support for AuditEvent
+ * https://www.hl7.org/fhir/r4/auditevent.html
+ */
+public class AuditEventMapper implements Mapper {
+
+    private static final Logger logger = java.util.logging.Logger.getLogger(AuditEventMapper.class.getName());
+    private static final String CLASSNAME = AuditEventMapper.class.getName();
+
+    //@formatter:off
```

Comment:
personally not a huge fan of all the formatter comments, but i understand why you add them.  nothing to change here, but maybe eventually we can settle on a common style (include them or not) :-)

Reply:
Sure - I hate hitting ctrl + f or having something accidentally formatted.